crates/valence_boss_bar/src/components.rs

```diff
@@ -73,4 +73,18 @@ pub struct BossBarFlags {
 }
 
 #[derive(Component)]
-pub struct BossBarViewers(pub Vec<Entity>);
\ No newline at end of file
+pub struct BossBarViewers {
+    pub current_viewers: Vec<Entity>,
```

Comment:
I keep track of the previous viewers to know which one were added/removed when this component is changed.